src/App.jsx

```diff
@@ -14,33 +16,112 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const IS_SORT_BY_ALPHABET = 'alphabet';
+const IS_SORT_BY_LENGTH = 'length';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+const getGoods = (goods, sortType, isReversed, isReseted) => {
+  let visibleGoods = [...goods];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (isReseted) {
+    return visibleGoods;
+  }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  if (sortType === IS_SORT_BY_ALPHABET) {
+    visibleGoods = visibleGoods.sort((a, b) => a.localeCompare(b));
+  }
+
+  if (sortType === IS_SORT_BY_LENGTH) {
+    visibleGoods = visibleGoods.sort((good1, good2) => {
+      if (good1.length === good2.length) {
+        return good1.localeCompare(good2);
+      }
+
+      return good1.length - good2.length;
+    });
+  }
+
+  if (isReversed) {
+    visibleGoods = visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+};
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+export const App = () => {
+  const [sortType, setSortType] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+  const [isReseted, setIsReseted] = useState(true);
+
+  const visibleGoods = getGoods(
+    goodsFromServer,
+    sortType,
+    isReversed,
+    isReseted,
+  );
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn('button', 'is-info', {
+            'is-light': sortType !== IS_SORT_BY_ALPHABET,
+          })}
+          onClick={() => {
+            setSortType(IS_SORT_BY_ALPHABET);
+            setIsReseted(false);
+          }}
+        >
+          Sort alphabetically
+        </button>
+        <button
+          type="button"
+          className={cn('button', 'is-success', {
+            'is-light': sortType !== IS_SORT_BY_LENGTH,
+          })}
+          onClick={() => {
+            setSortType(IS_SORT_BY_LENGTH);
+            setIsReseted(false);
+          }}
+        >
+          Sort by length
+        </button>
+        <button
+          type="button"
+          className={cn('button', 'is-warning', {
+            'is-light': !isReversed,
+          })}
+          onClick={() => {
+            setIsReversed(!isReversed);
+            setIsReseted(isReversed && sortType === '');
+          }}
+        >
+          Reverse
+        </button>
+        {(!isReseted || sortType !== '' || isReversed) && (
```

Comment:
- You don't need to compare with an empty string, just check if a value exists
- Create a variable for `sortType || isReversed` so you don't need to combine different logical operators